### PR TITLE
Emcee support

### PR DESCRIPTION
**Pull request by bocklund**
@richardotis Can you do a sanity check on the paramselect.fit() function paramselect.lnprob() function, the fit.py entry point, and the new stuff in utils? All the other changes should just be refactoring and should not have changed behavior.

### CODE REVIEW

**Comment by richardotis** [comment]
Regarding making ESPEI a package, you might find this section of the setuptools docs helpful: https://setuptools.readthedocs.io/en/latest/setuptools.html#automatic-script-creation

You can define a `console_scripts` entry point for fit.py called `espei`, and then let setuptools handle the cross platform details of generating the scripts. Then, as long as the ESPEI package is in the user's PATH (i.e., virtualenv activated if it's in a virtualenv), they can call `espei --help`, etc.

**Comment by bocklund** [comment]
It's set up like that in the master branch

File: espei/paramselect.py [hunk 1]
```diff
@@ -973,102 +397,19 @@ def bin_int_sort_key(x):
...
             if degree_polys[degree] != 0:
                 for syminter in symmetric_interactions:
-                    dbf.add_parameter('L', phase_name,
-                                      tuple(map(_to_tuple, syminter)), degree,
-                                      degree_polys[degree])
-    # Now fit ternary interactions
-
+                    dbf.add_parameter('L', phase_name, tuple(map(_to_tuple, syminter)), degree, degree_polys[degree])
+    # TODO: fit ternary interactions
     if hasattr(dbf, 'varcounter'):
```
Comment (richardotis) [inline espei/paramselect.py]: I think `dbf.varcounter` is dead code

File: espei/paramselect.py [hunk 2]
```diff
@@ -1077,113 +418,66 @@ def estimate_hyperplane(dbf, comps, phases, current_statevars, comp_dicts,
...
+def tieline_error(dbf, comps, current_phase, cond_dict, region_chemical_potentials, phase_flag,
+                  phase_models, parameters, debug_mode=False):
     if np.any(np.isnan(list(cond_dict.values()))):
         # We don't actually know the phase composition here, so we estimate it
         single_eqdata = calculate(dbf, comps, [current_phase],
                                   T=cond_dict[v.T], P=cond_dict[v.P],
-                                  model=phase_models, parameters=parameters,
-                                  pdens=10)
-        # print('SINGLE_EQDATA (UNKNOWN COMP)', single_eqdata)
+                                  model=phase_models, parameters=parameters, pdens=10)
```
Comment (richardotis) [inline espei/paramselect.py]: This was my mistake: running with `pdens` this low may actually cause problems for multi-sublattice solidus composition estimates. It's computing driving force using an unrefined point sample and then taking the minimum. A really coarse sample may lead to poor estimates.

File: espei/paramselect.py [hunk 3]
```diff
@@ -1503,107 +645,91 @@ def fit(input_fname, datasets, resume=None, scheduler=None, recfile=None,
...
+            np.savetxt(tracefile, sampler.flatchain)
+
+    # set up the MCMC run
+    # set up the initial parameters
+    # apply a Gaussian random to each parameter with std dev of 0.10*parameter
+    initial_parameters = np.array(initial_parameters)
+    ndim = len(initial_parameters)
+    nwalkers = 2*ndim # walkers must be of size (2n*ndim)
+    initial_walkers = np.tile(initial_parameters, (nwalkers, 1))
+    walkers = np.random.normal(initial_walkers, np.abs(initial_walkers*0.10))
```
Comment (richardotis) [inline espei/paramselect.py]: To try to limit stochasticity for debugging purposes I would use
```python
rng = np.random.RandomState(1769)
# ...
walkers = rng.normal(...)
```

I know emcee has its own source of randomness, but that can be addressed later.

File: espei/paramselect.py (hunk 2 above)
Comment (bocklund) [inline espei/paramselect.py]: What do you recommend?

Comment (richardotis) [inline espei/paramselect.py]: I might go with `pdens = 100`. Not sure about the effect on memory consumption.

File: espei/paramselect.py (hunk 1 above)
Comment (bocklund) [inline espei/paramselect.py]: Isn't this how the VV#### variables are named in full fits?

Comment (richardotis) [inline espei/paramselect.py]: Yes, you are correct. I had in mind it was for something else. Definitely should be kept.

**Pull request merged**